src/VSSetup.PowerShell/PowerShell/SelectInstanceCommand.cs

```diff
@@ -95,9 +97,10 @@ protected override void ProcessRecord()
             if (Product != null && Product.Any())
             {
                 // Select instances with no product or the specified product ID.
+                var patterns = Product.Select(product => GetPattern(product));
```

Comment:
Looks like the deferred execution of the Select could happen multiple times due to the usage in the where clause below.